test/e2e/kubevirt.go

```diff
@@ -870,160 +876,402 @@ passwd:
-	unstructuredVMIMigrations := &unstructured.UnstructuredList{}
-	unstructuredVMIMigrations.SetGroupVersionKind(schema.GroupVersionKind{
-		Group:   kubevirtv1.GroupVersion.Group,
-		Kind:    "VirtualMachineInstanceMigrationList",
-		Version: kubevirtv1.GroupVersion.Version,
-	})
+			virtualMachineInstance = resourceCommand{
+				description: "VirtualMachineInstance",
+				cmd: func() string {
+					var err error
+					vmi, err = fcosVMI(1, nil /*labels*/, nil /*annotations*/, nil /*nodeSelector*/, kubevirtv1.NetworkSource{
+						Multus: &kubevirtv1.MultusNetwork{
+							NetworkName: nad.Name,
+						},
+					}, butane)
+					Expect(err).ToNot(HaveOccurred())
+					createVirtualMachineInstance(vmi)
+					return vmi.Name
+				},
+			}
+			filterOutIPv6 = func(ips map[string][]string) map[string][]string {
+				filteredOutIPs := map[string][]string{}
+				for podName, podIPs := range ips {
+					for _, podIP := range podIPs {
+						if !utilnet.IsIPv6String(podIP) {
+							_, ok := filteredOutIPs[podName]
+							if !ok {
+								filteredOutIPs[podName] = []string{}
+							}
+							filteredOutIPs[podName] = append(filteredOutIPs[podName], podIP)
+						}
+					}
+				}
+				return filteredOutIPs
+			}
+		)
+		type testData struct {
+			description string
+			resource    resourceCommand
+			test        testCommand
+			topology    string
+		}
+		DescribeTable("should keep ip", func(td testData) {
+			By("Creating NetworkAttachmentDefinition")
+			nad = generateNAD(newNetworkAttachmentConfig(networkAttachmentConfigParams{
+				namespace:          namespace,
+				name:               "net1",
+				topology:           td.topology,
+				cidr:               strings.Join([]string{cidrIPv4, cidrIPv6}, ","),
+				allowPersistentIPs: true,
+			}))
+			Expect(crClient.Create(context.Background(), nad)).To(Succeed())
+			workerNodeList, err := fr.ClientSet.CoreV1().Nodes().List(context.Background(), metav1.ListOptions{LabelSelector: labels.FormatLabels(map[string]string{"node-role.kubernetes.io/worker": ""})})
+			Expect(err).ToNot(HaveOccurred())
+			selectedNodes = workerNodeList.Items
+			networkName := fmt.Sprintf("%s/%s", nad.Namespace, nad.Name)
+			prepareHTTPServerPods(map[string]string{
+				"k8s.v1.cni.cncf.io/networks": fmt.Sprintf(`[{"name": %q}]`, nad.Name),
+			}, checkPodHasIPsAtNetwork(networkName, 2 /*expectedNumberOfAddresses*/))
+
+			vmiName := td.resource.cmd()
+			vmi = &kubevirtv1.VirtualMachineInstance{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: namespace,
+					Name:      vmiName,
+				},
+			}
+			waitVirtualMachineInstanceReadiness(vmi)
+			Expect(crClient.Get(context.TODO(), crclient.ObjectKeyFromObject(vmi), vmi)).To(Succeed())
 
-	if err := client.List(context.Background(), unstructuredVMIMigrations); err != nil {
-		return nil, err
-	}
-	if len(unstructuredVMIMigrations.Items) == 0 {
-		return nil, fmt.Errorf("empty migration list")
-	}
+			step := by(vmi.Name, "Login to virtual machine for the first time")
```

Comment:
not here, but in some other places not changed by this PR  you have a unnecessary `fmt.Sprintf` on your `by` statements

Reply:
Will remove Sprintf from were is not really needed.